frontend/templates/ToolPage/ToolForm/ToolForm.jsx

```diff
@@ -38,41 +38,129 @@ const ToolForm = (props) => {
   const { register, control, handleSubmit, getValues, setValue, errors } =
     useWatchFields([]);
 
-  const handleSubmitMultiForm = async (values) => {
-    try {
-      const { files, ...toolData } = values;
-
-      dispatch(setResponse(null));
-
-      const updateData = Object.entries(toolData).map(([name, value]) => ({
-        name,
-        value,
-      }));
-      dispatch(setPrompt(values));
-      dispatch(setCommunicatorLoading(true));
-
-      const response = await submitPrompt(
-        {
-          tool_data: { tool_id: id, inputs: updateData },
-          type: 'tool',
-          user: {
-            id: userData?.id,
-            fullName: userData?.fullName,
-            email: userData?.email,
-          },
-        },
-        files
-      );
+  // Helper function to extract the YouTube video ID from the URL
+  const extractVideoId = (url) => {
+    const urlObj = new URL(url);
+    return urlObj.searchParams.get('v');
+  };
 
-      dispatch(setResponse(response?.data));
-      dispatch(setFormOpen(false));
-      dispatch(setCommunicatorLoading(false));
-    } catch (error) {
-      dispatch(setCommunicatorLoading(false));
-      handleOpenSnackBar(
-        ALERT_COLORS.ERROR,
-        error?.message || 'Couldn\u0027t send prompt'
-      );
+  // Helper function to parse ISO 8601 duration to seconds
+  const parseDuration = (isoDuration) => {
+    const match = isoDuration.match(/PT(\d+H)?(\d+M)?(\d+S)?/);
+    const hours = parseInt(match[1], 10) || 0;
+    const minutes = parseInt(match[2], 10) || 0;
+    const seconds = parseInt(match[3], 10) || 0;
+    return hours * 3600 + minutes * 60 + seconds;
+  };
+
+  const handleSubmitForm = async (values) => {
```

Comment:
Technically this check is done on the AI side, would like to know why this logic has been introduced